tools/Dashboard/DevHome.Dashboard/Services/IAdaptiveCardRenderingService.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System.Threading.Tasks;
+using AdaptiveCards.Rendering.WinUI3;
+
+namespace DevHome.Dashboard.Services;
+
+internal interface IAdaptiveCardRenderingService
+{
+    public Task<AdaptiveCardRenderer> GetRenderer();

```

Comment:
Hmm I wonder if its possible to use the interface `IAdaptiveCardRenderer`. instead of the concrete type AdaptiveCardRenderer. This way IAdaptiveCardRenderingService can be completely mocked without having to create the winrt AdaptiveCardRenderer object.  I think that would work but I could be wrong, let me know what you think

Reply:
Good call out but IAdaptiveCardRenderer isn't accessible (it's `internal`) so I don't think we can. I'm not sure if it's possible/easy to make a change [over](https://github.com/microsoft/AdaptiveCards/blob/f003fe18681a0ba428f32ecadad3df7fc63529c4/source/uwp/SharedRenderer/lib/AdaptiveCardRendererComponent.h) [there](https://github.com/microsoft/AdaptiveCards/blob/f003fe18681a0ba428f32ecadad3df7fc63529c4/source/uwp/winui3/Renderer/idl/AdaptiveCards.Rendering.WinUI3.idl#L38) to make this possible.